test/connection.test.js

```diff
@@ -1066,24 +1095,47 @@ test('transaction', async () => {
     },
     {
       error: null,
-      result: [
-        {
-          slot: 0,
-          status: {Ok: null},
+      result: {
+        context: {
+          slot: 11,
         },
-        null,
-      ],
+        value: [
+          {
+            slot: 0,
+            confirmations: 11,
+            status: {Ok: null},
+          },
+          null,
+        ],
+      },
     },
   ]);
 
-  const responses = await connection.getSignatureStatusBatch([
-    signature,
-    unprocessedSignature,
-  ]);
+  const responses = (
+    await connection.getSignatureStatusBatch([signature, unprocessedSignature])
+  ).value;
   expect(responses.length).toEqual(2);
-  expect(responses[0]).toEqual(response);
+
+  const firstResponse = responses[0];
   expect(responses[1]).toBeNull();
 
+  if (firstResponse === null) {
+    expect(firstResponse).not.toBeNull();
+    return;
+  }
+
+  expect(firstResponse.slot).toBeGreaterThanOrEqual(response.slot);
+  expect(firstResponse.status).toEqual(response.status);
+
+  if (typeof firstResponse.confirmations !== 'number') {
+    expect(typeof firstResponse.confirmations).toEqual('number');
+    return;
+  }
+
+  expect(firstResponse.confirmations).toBeGreaterThanOrEqual(
+    responseConfirmations,
+  );
+
```

Comment:
What do you think about adding a test case here for `confirmations: null`? Should be able to loop on getSignatureStatus with commitment: 'max' to know when the transaction is rooted for the live test.

Seems like a pretty easy add, and it'd be nice to have that tested in an integration test somewhere, at least.

Reply:
Yes, great call!